File: 14.longest-common-prefix.py
#
# @lc app=leetcode id=14 lang=python3
#
# [14] Longest Common Prefix
#

# @lc code=start
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:            
    # if strs is empty, return empty string
        if not strs:
            return ""

        # if strs has only one element, return that element
        if len(strs) == 1:
            return strs[0]

        # find the shortest string in strs
        shortest = min(strs, key=len)

        # compare each character of shortest with each character of the other strings
        for i, char in enumerate(shortest):
            for other in strs:
                if other[i] != char:
                    return shortest[:i]

        return shortest
# @lc code=end

File: 14.longest-common-prefix.cpp
/*
 * @lc app=leetcode id=14 lang=cpp
 *
 * [14] Longest Common Prefix
 */

// @lc code=start
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string prefix = "";
        
        if (strs.size() == 0) {
            return prefix;
        }
        
        for (int i = 0; i < strs[0].length(); i++) {
            char c = strs[0][i];
            
            for (int j = 1; j < strs.size(); j++) {
                if (i >= strs[j].length() || strs[j][i] != c) {
                    return prefix;
                }
            }
            
            prefix += c;
        }
        
        return prefix;
    }
};
// @lc code=end

